### PR TITLE
Remove Sonarqube

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a summary of the changes.)

The goal is to remove Sonarqube. All code quality checks are done by Code Climate which is already set up. Code coverage is reported by Code Cov for now (eventually we may configure Code Climate for this).

### In OpenShift:

- Remove `sonarqube` and `postgresql-sonarqube` Deployment Configs.
  - removes `sonarqube` ImageStream.
  - removes `sonarqube` and `postgresql-sonarqube` Services.
  - removes `sonarqube` Route.

- Remove if not already removed - `sonarqube` BuildConfig.
- Remove if not already removed -  `postgresql-sonarqube-data` and `sonarqube-data` PersistentVolumeClaims.
- Remove if not already removed -  `postgresql-sonar-password`, `sonarqube-admin-password` and `sonarqube-token` Secrets.

[](https://hamzamohdzubair.github.io/redant/)

[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a list of Jira tickets and or Issues this PR resolves.)

- [PIMS-135](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-135)

[](https://hamzamohdzubair.github.io/redant/)

[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please include a list of known issues if there are any.)

- None listed.

## Type of change
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please delete options that are not relevant.)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency updates (may include refactoring of a dependencies functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
[//]: # (THESE COMMENTS ARE HIDDEN WHEN THE PR IS PUBLISHED, no need to remove them.)
[//]: # (Please describe the tests that you ran to verify your changes.)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by BradyMitch** [comment]
I was wondering if OWASP scanner would be a problem because in `openshift/4.0/templates/tekton/tasks/owasp-scanner.yaml` the TARGET_URL parma says "Host URL where the sonarqube server is running" but it seems that in both dev and test pipelines this parameter is just the regular web app url.

File: .github/workflows/api-dotnetcore.yml [hunk 1]
```diff
@@ -96,66 +91,3 @@ jobs:
           env_vars: C#
           # Specify whether or not CI build should fail if Codecov runs into an error during upload
           fail_ci_if_error: true
-
-      - name: SonarScanner for .NET 6 with pull request decoration support
```
Comment (BradyMitch) [inline .github/workflows/api-dotnetcore.yml:100]: Removed - The results of this scan are used to generate a comment in the pull request thread.

File: .github/workflows/app-react.yml [hunk 2]
```diff
@@ -55,65 +50,3 @@ jobs:
           env_vars: javascript
           # Specify whether or not CI build should fail if Codecov runs into an error during upload
           fail_ci_if_error: true
-
-      - name: SonarQube Scan
```
Comment (BradyMitch) [inline .github/workflows/app-react.yml:59]: Removed - The results of this scan are used to generate a comment in the pull request thread.

File: frontend/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "cypress:e2e-dev": "cypress run --e2e --browser chrome --config baseUrl=\"https://pims-dev.apps.silver.devops.gov.bc.ca\" --env auth_client_id=\"pims-frontend-4391\"",
     "cypress:e2e-test": "cypress run --e2e --browser chrome --config baseUrl=\"https://pims-test.apps.silver.devops.gov.bc.ca\" --env auth_base_url=\"https://test.loginproxy.gov.bc.ca/auth\",auth_client_id=\"pims-frontend-4391\"",
     "cypress:e2e-prod": "cypress run --e2e --browser chrome --config baseUrl=\"https://pims.gov.bc.ca\" --env auth_base_url=\"https://loginproxy.gov.bc.ca/auth\",auth_client_id=\"pims-frontend-4391\"",
-    "coverage": "react-scripts test --ci --coverage --testResultsProcessor jest-sonar-reporter --watchAll=false --runInBand",
```
Comment (BradyMitch) [inline frontend/package.json:18]: This was formatting the test coverage in a way that sonarqube could understand.

File: openshift/4.0/player.sh [hunk 4]
```diff
@@ -27,15 +24,6 @@ case "${1:-}" in
     deploy)
         ./oc-deploy.sh ${ARGV}
         ;;
-    scan)
-        ./oc-sonar-scan.sh ${ARGV}
-        ;;
-    scan-dotnet)
```
Comment (BradyMitch) [inline openshift/4.0/player.sh:33]: These were never used

File: openshift/4.0/templates/sonarqube/sonarqube-postgresql-template.yaml [hunk 5]
```diff
@@ -1,404 +0,0 @@
-apiVersion: v1
```
Comment (BradyMitch) [inline openshift/4.0/templates/sonarqube/sonarqube-postgresql-template.yaml:1]: Template for deploying SonarQube.

Created:
	Service: sonarqube
	Route: sonarqube
	Secret: sonarqube-admin-password
	Secret: postgresql-sonar-password
	ImageStream: sonarqube
	DeploymentConfig: sonarqube
	Service: postgresql-sonarqube
	DeploymentConfig: postgresql-sonarqube
	PersistentVolumeClaim: postgresql-sonarqube-data
	PersistentVolumeClaim: sonarqube-data
	NetworkSecurityPolicy: pod-to-pod-permit-${NAME}-${SOURCE_ROLE}-${DESTINATION_ROLE}

File: openshift/4.0/templates/sonarqube/deploy.yaml [hunk 6]
```diff
@@ -1,163 +0,0 @@
-apiVersion: v1
```
Comment (BradyMitch) [inline openshift/4.0/templates/sonarqube/deploy.yaml:1]: Defines a set of objects needed to deploy and run a SonarQube application.

Created:
	ImageStream - sonarqube
	DeploymentConfig - sonarqube
	PersistentVolumeClaim
	Service - sonarqube
	Route - sonarqube

File: openshift/4.0/scripts/oc-sonar-dotnet.sh [hunk 7]
```diff
@@ -1,115 +0,0 @@
-#!/bin/bash
-#
```
Comment (BradyMitch) [inline openshift/4.0/scripts/oc-sonar-dotnet.sh:2]: This wasn't being used anywhere in the repo.

Performs a Sonar scan on a .NET Core codebase. 
It cleans up any previous test runs, builds the code, 
runs unit tests with coverage reporting, 
and finally performs the Sonar scan.

File: openshift/4.0/scripts/oc-sonar-scan.sh [hunk 8]
```diff
@@ -1,68 +0,0 @@
-#!/bin/bash
-#
```
Comment (BradyMitch) [inline openshift/4.0/scripts/oc-sonar-scan.sh:2]: This wasn't used anywhere in the repo.

Runs a Sonar scan for a JavaScript/TypeScript codebase, including unit tests and coverage.

**Review by LawrenceLau2020 — APPROVED** [review]

**Pull request merged**